docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -619,6 +619,97 @@ You can also add any number of `else` blocks:
 {/if}
 ----
 
+[[when_section]]
+==== When/Switch Section
+
+This section is similar to Java's `switch` or Kotlin's `when` constructs.
```

Comment:
In general I don't like when there's two names for the same thing and I think `swich/case` should be enough. But if you say that's what Kotlin uses, then perhaps that's a good justification for having both.